calculator.c

```diff
@@ -0,0 +1,117 @@
+#include <stdio.h>
+#include <ctype.h>
+#include <stdlib.h>
+#define MAX_SIZE 100
+
+int operation(int a, int b, char op)
+{
+    switch (op)
+    {
+    case '+':
+        return a + b;
+    case '-':
+        return a - b;
+    case '*':
+        return a * b;
+    case '/':
+        if (b == 0)
+        {
+            printf("Error: Division by zero.\n");
+            exit(1);
+        }
+        return a / b;
+    default:
+        return 0;
+    }
+}
+
+int precedence(char op)
+{
+    if (op == '+' || op == '-')
+        return 1;
+    if (op == '*' || op == '/')
+        return 2;
+    return 0;
+}
+
+int evaluate(char *exp)
+{
+    int stack1[MAX_SIZE], top1 = -1;
```

Comment:
Can variable names match below standards, take care of this at all places:

Use descriptive names: Choose meaningful names that reflect the purpose of the variable.
Use snake_case: Use lowercase letters with underscores to separate words.
ex :
int total_count;
float average_score;
char student_name[50];
Global variables:

Prefix with g_ or global_ to differentiate from local variables.
ex :
int g_total_count;
Constants:

Use all uppercase letters with underscores to separate words.
ex:
#define MAX_BUFFER_SIZE 1024
const float PI = 3.14159